include/lsst/afw/cameraGeom/TransformMap.h

```diff
@@ -0,0 +1,213 @@
+/*
+ * LSST Data Management System
+ * Copyright 2014 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#if !defined(LSST_AFW_CAMERAGEOM_TRANSFORMMAP_H)
+#define LSST_AFW_CAMERAGEOM_TRANSFORMMAP_H
+
+#include <vector>
+#include <map>
+#include <unordered_map>
+
+#include "boost/iterator/transform_iterator.hpp"
+#include "astshim/FrameSet.h"
+
+#include "lsst/afw/cameraGeom/CameraSys.h"
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/geom/Transform.h"
+
+namespace lsst {
+namespace afw {
+namespace cameraGeom {
+
+/**
+ * Functor for boost::transform_iterator: given an entry in a std::map or unordered_map, return the key
+ */
+struct GetKey {
+    CameraSys const &operator()(std::pair<const CameraSys, int> const &p) const { return p.first; };
+};
+
+/**
+ * A registry of 2-dimensional coordinate transforms for a specific camera.
+ *
+ * Represents the interrelationships between camera coordinate systems for a
+ * particular camera. It can be seen as a mapping between a pair of CameraSys
+ * and a Transform between them, but does not conform to either the C++ map or
+ * Python dictionary APIs.
+ *
+ * TransformMap supports transforming between any two supported CameraSys using
+ * the @ref transform methods, even if no explicit transform was provided.
+ *
+ * @exceptsafe Unless otherwise specified, all methods guarantee only basic
+ *             exception safety.
+ */
+class TransformMap final {
+public:
+    using Transform = afw::geom::Transform<geom::Point2Endpoint, geom::Point2Endpoint>;
+    using Transforms = std::unordered_map<CameraSys, std::shared_ptr<Transform>>;
+    using CameraSysFrameIdMap = std::unordered_map<CameraSys, int>;
+    using CameraSysIterator = boost::transform_iterator<GetKey, CameraSysFrameIdMap::const_iterator>;
+
+    /**
+     * Define a set of camera transforms.
+     *
+     * @param reference Coordinate system to which `transforms` converts.
+     * @param transforms A map whose keys are coordinate systems, and whose
+     *                   values point to Transforms that convert from
+     *                   `reference` to the corresponding key. All Transforms
+     *                   must be invertible.
```

Comment:
The docs for the parameters are inconsistent: docs for `reference` say that all elements in `transforms` convert *to* `reference`, while docs for `transforms` say all elements in `transforms` convert *from* `reference`.